client/client-lib/src/lib.rs

```diff
@@ -110,7 +110,7 @@ impl From<GatewayClientConfig> for LightningGateway {
 }
 
 pub struct Client<C> {
-    context: Arc<OwnedClientContext<C>>,
+    inner: Arc<(C, ClientContext)>,
```

Comment:
Unrelated to the PR: why is this an `Arc`? It would be nice to give the config and context a name and not just an index.

Reply:
83ce8dff4f86c418fe29cefee1c53b8f9490d615: `Arc` removed and we have fields `config` and `context`.